client/my-sites/plans/jetpack-plans/jetpack-complete-page/components/cta-buttons/cta-buttons.tsx

```diff
@@ -0,0 +1,45 @@
+import { PLAN_JETPACK_COMPLETE } from '@automattic/calypso-products';
+import { useTranslate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import Button from 'calypso/components/forms/form-button';
+import getSelectedSiteSlug from 'calypso/state/ui/selectors/get-selected-site-slug';
+
+import './style.scss';
+
+const CtaButtons = () => {
+	const translate = useTranslate();
+	const planToOffer = PLAN_JETPACK_COMPLETE;
+	const siteSlug = useSelector( getSelectedSiteSlug );
+
+	// handle no site selected
+	const getCompleteURL = () => {
+		if ( siteSlug ) {
+			return `/checkout/${ siteSlug }/${ planToOffer }`;
+		}
+		return `https://wordpress.com/checkout/${ planToOffer }`;
+	};
```

Comment:
Depending on whether or not we have a site slug, someone testing in an environment outside of production or staging (e.g., `http://jetpack.cloud.localhost:3000`) could be sent to either of two different domains, which could be a big source of frustration if they don't notice the difference.

If we're satisfied with excluding the domain for one case, let's do it for both:

```suggestion
	const getCompleteURL = () => {
		if ( siteSlug ) {
			return `/checkout/${ siteSlug }/${ planToOffer }`;
		}
		return `/checkout/${ planToOffer }`;
	};
```

---

Second, lower priority: using ternaries, we could make this a variable instead of a function 🙂 Using exactly what you've written, here's an example:

```js
const completeURL = siteSlug
	? `/checkout/${ siteSlug }/${ planToOffer }`
	: `https://wordpress.com/checkout/${ planToOffer }`;
```

---

Third, since you've mentioned that you're studying hooks like `useMemo`: it wouldn't help much in this case since the logic is very inexpensive and the component itself doesn't have a complex enough state to cause it to re-render often, but a memoized version of exactly the code you have here might look like so:

```js
const completeURL = useMemo( () => {
	if ( siteSlug ) {
			return `/checkout/${ siteSlug }/${ planToOffer }`;
		}
		return `https://wordpress.com/checkout/${ planToOffer }`;
}, [ siteSlug, planToOffer ] );
```

Reply:
Alternatively, we could probably also use [buildCheckoutUrl](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans/jetpack-plans/get-purchase-url-callback.ts#L27) function. Though, I'm also ok with manual links.

Reply:
@andrii-lysenko That's a great point -- I wasn't aware of that function, but re-using well-worn code gives us an extra level of confidence that what we're writing will be correct and require less maintenance.

Reply:
Question on the finer points of buildCheckoutUrl.  Even though it appears to be able to handle if you don't pass a site to it, siteId is typed as string.  I'm not sure what to pass if there's no ID to make the linter not whine at me for passing in something that could have a null value.

Reply:
@TheSteveK good question. So it could be either `url = buildCheckoutURL( siteSlug || '', ...)` or `url = siteSlug ? buildCheckoutURL( siteSlug,) : ''`. It should work as we should have siteId always here. However, I see also an option to modify the `buildCheckoutURL` so it could work without necessarily passing `siteSlug`.